app/src/main/java/com/example/playlistmaker/SearchActivity.kt

```diff
@@ -50,22 +60,109 @@ class SearchActivity : AppCompatActivity() {
-        text = findViewById(R.id.status_text)
+        statusText = findViewById(R.id.status_text)
         updateButton = findViewById(R.id.update_button)
+        beenSearchedTitle = findViewById(R.id.been_searched_title)
         backButtonClickAttach()
         searchBarTextWatcherAttach()
         searchBarSetActionDone()//temporary use only
         clearTextAttach()
         startUpViewHolder()
         findViewById<TextView>(R.id.update_button).setOnClickListener { sendRequest() }
+        clearHistoryButtonClickAttach()
     }
 
-    private fun searchBarSetActionDone() {
-        searchBar.setOnEditorActionListener { _, actionId, _ ->
-            if (actionId == EditorInfo.IME_ACTION_DONE) {
-                sendRequest()
+    override fun onPause() {
+        super.onPause()
+        App.history.saveToVault()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        if (searchBar.text.isNullOrEmpty()) xMark.visibility = View.GONE
+        App.history.getFromVault()
+        if (App.history.tracks.isEmpty()) showLayout(Layout.CleanHistory)
+        else showLayout(Layout.History)
+    }
+
+    override fun onSaveInstanceState(outState: Bundle, outPersistentState: PersistableBundle) {
+        super.onSaveInstanceState(outState, outPersistentState)
+        outState.putString(SEARCH_PROMPT, searchPromptString)
+    }
+
+    override fun onRestoreInstanceState(
+        savedInstanceState: Bundle?,
+        persistentState: PersistableBundle?
+    ) {
+        super.onRestoreInstanceState(savedInstanceState, persistentState)
+        searchBar.setText(
+            savedInstanceState?.getString(SEARCH_PROMPT)
+                ?: SEARCH_DEF
+        )
+    }
+
+    fun showLayout(layout: Layout) {
+        when (layout) {
+            Layout.SearchGot -> {
+                beenSearchedTitle.visibility = View.GONE
+                clearHistoryButton.visibility = View.GONE
+                placeholderFrame.visibility = View.GONE
+                if (trackListRecyclerView.adapter != tracksAdapter) trackListRecyclerView.swapAdapter(
+                    tracksAdapter,
+                    true
+                )
+                trackListRecyclerView.visibility = View.VISIBLE
+            }
+
+            Layout.SearchIsEmpty -> {
+                beenSearchedTitle.visibility = View.GONE
+                clearHistoryButton.visibility = View.GONE
+                updateButton.visibility = View.GONE
+                statusText.text = this.getString(R.string.search_status_nothing)
+                statusImageView.setImageDrawable(
+                    AppCompatResources.getDrawable(
+                        this,
+                        R.drawable.image_sad_smile_mus
+                    )
+                )
+                placeholderFrame.visibility = View.VISIBLE
+            }
+
+            Layout.Error -> {
+                beenSearchedTitle.visibility = View.GONE
```

Comment:
Есть удобный экстеншен `isVisible`:
```suggestion
                beenSearchedTitle.isVisible= false
```